app/helpers/application_helper.rb

```diff
@@ -192,6 +192,11 @@ def auto_complete_search(name, val, options = {})
     text_field_tag(name, val, options)
   end
 
+  def sort(field, permitted: [], **kwargs)
+    kwargs[:url_options] ||= params.permit(permitted + [:locale, :search])
+    super(field, kwargs)
```

Comment:
OK, I had a vague memory that we had similar problem with will paginate, but can't remember the root cause.. OK from my side.